src/sentry/integrations/slack/event_endpoint.py

```diff
@@ -19,27 +20,37 @@
 
 # XXX(dcramer): this could be more tightly bound to our configured domain,
 # but slack limits what we can unfurl anyways so its probably safe
-_link_regexp = re.compile(r"^https?\://[^/]+/[^/]+/[^/]+/(issues|incidents)/(\d+)")
+_link_regexp = re.compile(
+    r"^https?\://[^/]+/[^/]+/[^/]+/(issues|incidents)/(\d+)(?:/events/(\w+))?"
+)
 _org_slug_regexp = re.compile(r"^https?\://[^/]+/organizations/([^/]+)/")
 
 
-def unfurl_issues(integration, issue_map):
-    results = {
+def unfurl_issues(integration, url_by_issue_id, event_id_by_url=None):
```

Comment:
I think this function could benefit from a doc string, its kind of confusing to follow the logic and figure out why you have one dict with the key as the url and the other mapping the event id to the url